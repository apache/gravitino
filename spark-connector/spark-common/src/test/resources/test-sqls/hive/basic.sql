
-- If not specify location explicitly, the location will be hdfs://127.0.0.1:9000/xx, which is unaccessable for spark
CREATE DATABASE IF NOT EXISTS t_hive LOCATION '/user/hive/warehouse-spark-test/t_hive';
USE t_hive;

CREATE TABLE IF NOT EXISTS employees (
    id INT,
    name STRING,
    age INT
)
PARTITIONED BY (department STRING)
STORED AS PARQUET;

INSERT OVERWRITE TABLE employees PARTITION(department='Engineering') VALUES (1, 'John Doe', 30), (2, 'Jane Smith', 28);
INSERT OVERWRITE TABLE employees PARTITION(department='Marketing') VALUES (3, 'Mike Brown', 32);

SELECT * FROM employees WHERE department = 'Engineering';
