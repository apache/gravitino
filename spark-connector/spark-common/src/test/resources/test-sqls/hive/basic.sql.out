-- Automatically generated by Gravitino Spark SQL test
-- !query
CREATE DATABASE IF NOT EXISTS t_hive LOCATION '/user/hive/warehouse-spark-test/t_hive'
-- !query schema
struct<>
-- !query output



-- !query
USE t_hive
-- !query schema
struct<>
-- !query output



-- !query
CREATE TABLE IF NOT EXISTS employees (
    id INT,
    name STRING,
    age INT
)
PARTITIONED BY (department STRING)
STORED AS PARQUET
-- !query schema
struct<>
-- !query output



-- !query
INSERT OVERWRITE TABLE employees PARTITION(department='Engineering') VALUES (1, 'John Doe', 30), (2, 'Jane Smith', 28)
-- !query schema
struct<>
-- !query output



-- !query
INSERT OVERWRITE TABLE employees PARTITION(department='Marketing') VALUES (3, 'Mike Brown', 32)
-- !query schema
struct<>
-- !query output



-- !query
SELECT * FROM employees WHERE department = 'Engineering'
-- !query schema
struct<id:int,name:string,age:int,department:string>
-- !query output
1	John Doe	30	Engineering
2	Jane Smith	28	Engineering
